### PR TITLE
Update sd8887 driver to latest and port to Linux 5.15

### PR DESCRIPTION
**Pull request by majorz**
This Overwrites all files from the latest available source and firmware release.

Additional patches are added in order to support Linux 5.15.

Change-type: minor
Signed-off-by: Zahari Petkov <zahari@balena.io>

### CODE REVIEW

File: software/drivers/sd8887/src/wlan/mapp/uaputl/uaputl.h [hunk 1]
```diff
@@ -802,6 +802,7 @@ typedef enum {
 	GWK_CIPHER,
 	RESTRICT_CLIENT_MODE,
 	AKM_SUITE,
+	WACPMODE
```
Comment (alexgg) [inline software/drivers/sd8887/src/wlan/mapp/uaputl/uaputl.h:805]: @majorz what does WACP stand for?

Comment (majorz) [reply software/drivers/sd8887/src/wlan/mapp/uaputl/uaputl.h:805]: @alexgg this I believe is compatibility feature with Apple CarPlay. That commit in particular came upstream. I only did the ones that start with "Port".

Comment (majorz) [reply software/drivers/sd8887/src/wlan/mapp/uaputl/uaputl.h:805]: I squashed now the commits that came upstream. Now it is cleaner.

Comment (alexgg) [reply software/drivers/sd8887/src/wlan/mapp/uaputl/uaputl.h:805]: It is cleaner @majorz, thanks

**Review by alexgg — APPROVED** [review]
lgtm

**Pull request merged**